src/components/modules/InfoBox/index.tsx

```diff
@@ -0,0 +1,183 @@
+
+        this.setState({
+            readText: toggleText
+        })
+    }
+
+    // On Layout
+    // Get the dimensions of the element holding the tag description. Use that number to help determine height of the entire InfoBox module. Will need to add additional px to the number obtained by this function.
+    // Get the window width. Use that number in conjuction with the tag description dims to determine height of background image. Will need to subtract pts.
+    getHeight = (e: any) => {
+        let w = e.nativeEvent.layout.width;
+        let h = e.nativeEvent.layout.height;
+        let windowWidth = Dimensions.get('window').width;
+        let imgHeight = (windowWidth * h)/w; 
+        // Convert background height to react native's units of measurement for images
+        let finalImgHeight = (imgHeight * 100)/windowWidth; 
+
+        // On initial render, set state for h x w of description box, screen width, and background image height
+        if (this.state.imgHeight === 0) {
+            this.setState({ 
+
+                width: w,
+                height: h,
+                windowWidth: windowWidth,
+                imgHeight: finalImgHeight,
+            })
+        } else { // When toggling 'read more/read less' just update h x w for description box.
+            this.setState({ 
+                width: w,
+                height: h,
+            })
+        }
+    }
+
+    renderDesktop = () => {
+        const { data } = this.props;
+        return (
+            <ImageBackground source={{uri: data[0].imageSrc}} 
+            style={[styles.imageDesktop]}> 
+                <View style={[styles.infobox, styles.infoboxDesktop]}>
+                    <ModuleBox offsetDirection={OFFSET_DIRECTION.RIGHT} patternColor={COLOR_MAP.PURPLE} backgroundColor={'transparent'} style={styles.moduleBoxDesktop}>
+                        <View>
+                            <Text style={styles.title}>{Capitalize(data[0].tag)}</Text>
+                            <HtmlText html={data[0].description} css={htmlTextStyle} />
+                        </View>
+                    </ModuleBox>
+                </View>
+            </ImageBackground>
+        )
+    }
+
+    renderMobile = (isTablet: any) => {
+        const { data } = this.props;
+        const { readText, height, windowWidth, imgHeight } = this.state;
+        // Determine image height for device
+        let imageHeight = isTablet ? imgHeight : (windowWidth > 350 ? Math.floor(imgHeight - 45) : Math.floor(imgHeight - 80));
+        let closingPTag = '</p>';
+        let indexOfFirstClosingPTag = data[0].description.indexOf(closingPTag);
+        // Determine first paragraph of tag descritption
+        let descriptionSubstring = data[0].description.substring(0, indexOfFirstClosingPTag + closingPTag.length);
+        let toggleDescription = readText === 'Read More' ? descriptionSubstring : data[0].description;
```

Comment:
can you move all the description substring logic to its own function? The call would look something like:
`const toggleDescription = isExpanded ? description : getDescriptionSubstring(description);`

Reply:
Moved substring logic to its own function.